src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/HearingPartService.java

```diff
@@ -64,16 +66,26 @@
     }
 
     public List<HearingPartResponse> getAllHearingPartsThat(Boolean areListed) {
-        List<HearingPart> hearingParts;
+        List<HearingPart> hearingParts = hearingPartRepository.findAll();
```

Comment:
you are loading all hearing parts from db to java? and then filtering them with steam?

Reply:
@r-kruk-kainos  have you tried to add native query in the repository ?